tests/unittest/compiler/test_fuse_split_cat.py

```diff
@@ -0,0 +1,157 @@
+        concatenate_3._attrs["name"] = "output_0"
+        concatenate_3._attrs["is_output"] = True
+        # Compile
+        model = compile_model(
+            concatenate_3,
+            detect_target(),
+            "./tmp",
+            self._testMethodName
+        )
+        # Check that split was removed
+        self.assertFalse(graph_has_op(model.debug_sorted_graph, "split"))
+        # Run
+        input_1 = get_random_torch_tensor((M.value(), N.value()), dtype=dtype)
+        # Compare
+        split_pt = torch.split(input_1, int(M.value()/2), 0)
+        y_pt = torch.cat(
+            [split_pt[1], split_pt[0]],
+            0,
+        )
+        y_ait = torch.empty_like(y_pt)
+        model.run_with_tensors(
+            {"input_1": input_1},
+            [y_ait],
+        )
+        torch.testing.assert_close(y_ait, y_pt, atol=0, rtol=0)
+
+    def test_fuse_split_cat_dim1(self):
+        dtype = "float16"
+        M = IntImm(512)
+        N = IntImm(512)
+
+        input_1 = Tensor(
+            shape=[M, N],
+            name="input_1",
+            is_input=True,
+        )
+        split_2 = ops.split()(
+            input_1, int(N.value()/2), 1
+        )
+        concatenate_3 = ops.concatenate()(split_2[::-1], 1)
+
+        # Set outputs
+        concatenate_3._attrs["name"] = "output_0"
+        concatenate_3._attrs["is_output"] = True
+        # Compile
+        model = compile_model(
+            concatenate_3,
+            detect_target(),
+            "./tmp",
+            self._testMethodName
+        )
+        # Check that split was removed
+        self.assertFalse(graph_has_op(model.debug_sorted_graph, "split"))
+        # Run
+        input_1 = get_random_torch_tensor((M.value(), N.value()), dtype=dtype)
+        # Compare
+        split_pt = torch.split(input_1, int(N.value()/2), 1)
+        y_pt = torch.cat(
+            split_pt[::-1],
+            1,
+        )
+        y_ait = torch.empty_like(y_pt)
+        model.run_with_tensors(
+            {"input_1": input_1},
+            [y_ait],
+        )
+        torch.testing.assert_close(y_ait, y_pt, atol=0, rtol=0)
+
+
+    def test_fuse_split_cat_different_dims(self):
+        """The case of splitting and then catting on different dims is not
+        expected to be optimized currently."""
+        dtype = "float16"
+        M = IntImm(512)
+        N = IntImm(512)
+
+        input_1 = Tensor(
+            shape=[M, N],
+            name="input_1",
+            is_input=True,
+        )
+        split_2 = ops.split()(
+            input_1, int(M.value()/2), 0
+        )
+        concatenate_3 = ops.concatenate()(split_2[::-1], 1)
+
+        # Set outputs
+        concatenate_3._attrs["name"] = "output_0"
+        concatenate_3._attrs["is_output"] = True
+        # Compile
+        model = compile_model(
+            concatenate_3,
+            detect_target(),
+            "./tmp",
+            self._testMethodName
+        )
+        # Run
```

Comment:
nit - can we add a check to ensure the existence of the split op? Thanks.

Reply:
Yes, can add that check. But this particular case seems to pass both with and without the optimization, so a future expansion of this optimization would break that check.

Reply:
That would be fine. We just want to avoid surprise such as we accidentally removed the split op.